pkg/openSearch/openSearchQuery/compareHandler.go

```diff
@@ -79,16 +79,35 @@ func handleCompareOperatorContainsDifferent(fieldName string, fieldKeys []string
 // HandleCompareOperatorBeginsWith handles begins with
 func HandleCompareOperatorBeginsWith(fieldName string, fieldKeys []string, fieldValue any, querySettings *QuerySettings) esquery.Mappable {
 	// for list of values
+	if querySettings.WildcardArrays != nil &&
+		querySettings.WildcardArrays[fieldName] {
+		return handleCompareOperatorBeginsWithDifferent(fieldName, fieldKeys, fieldValue, querySettings)
+	} else {
+		if values, ok := fieldValue.([]interface{}); ok {
+			return esquery.Bool().
+				Should(
+					lo.Map[interface{}, esquery.Mappable](values, func(value interface{}, _ int) esquery.Mappable {
+						return esquery.Prefix(fieldName+".keyword", valueToString(value))
+					})...,
+				).
+				MinimumShouldMatch(1)
+		} else { // for single values
+			return esquery.Prefix(fieldName+".keyword", valueToString(fieldValue))
+		}
+	}
```

Comment:
As far as I see, the only difference to `handleCompareOperatorBeginsWithDifferent` is that `esquery.Prefix` is called with  `fieldName+".keyword"` instead of `fieldName`. So you could just call `handleCompareOperatorBeginsWithDifferent` just with the different arguments, depending on the the `querySettings.WildcardArrays[fieldName]` entry.

(Another difference is `fieldValue.(string)` in `handleCompareOperatorBeginsWithDifferent` vs `valueToString(fieldValue)`, but I think it should always be the latter.